plugin/pkg/shoot/validator/admission.go

```diff
@@ -574,6 +590,41 @@ func (v *ValidateShoot) Admit(a admission.Attributes, o admission.ObjectInterfac
 	return nil
 }
 
+// validateMetaUpdateOnDeletedSeed validates updates to a Shoot's metadata in the case that the referenced Seed is
+// already marked for deletion. In this case it should not be allowed to modify any annotations except adding the
+// deletion confirmation annotation and to modify the Finalizers array. These operations are required to delete
+// Shoots to cleanup a Seed which is marked for deletion
+// It returns true, if the metadata update can be admitted, otherwise false.
+func validateMetaUpdateOnDeletedSeed(newMeta, oldMeta *metav1.ObjectMeta) bool {
```

Comment:
Actually the function has no knowledge of a `seed` object. Please adjust the doc string and the function name to the job it does.